code/graphics/2d.cpp

```diff
@@ -1877,6 +1877,24 @@ void gr_set_color_fast(const color *dst)
 	gr_screen.current_color = *dst;
 }
 
+//Compares the RGBA values of two colors. Returns true if the colors are identical
+bool gr_compare_color_values(color* clr1, color* clr2)
```

Comment:
This function signature should probably be changed to recieve ``const color&`` instead of ``color*``.
The constness, for one, allows much more flexible function use, and should generally be adhered to everywhere. In fact, I am surprised that clang-tidy doesn't flag this. I thought I'd set it to complain about missing constness in function parameters.
The reference instead of the pointer is a modernization we should work towards in most parts of the codebase, as it significantly reduces the issue of potential null pointers, especially when there is no need to take things as pointers (which, really, is mostly just old ``const char*`` strings, as well as optional return values as optional references are disallowed).

Reply:
Easy enough. Done